### ISSUE TITLE
Generate WebID Cert

### ISSUE THREAD

**Issue by sible777** [sible777/node-webid#2]
This is a copy of issue [#7](https://github.com/linkeddata/node-webid/issues/7) from the original repo.

The webid generate method should have the following signature.
```javascript
generate(options, callback) {
    // Do some magic in here
}
```

**Comment by sible777** [comment]
Mentioning you @nicola, so that you can stay up to date on this :+1:

**Comment by sible777** [comment]
Working on [issue #3](https://github.com/sible777/node-webid/issues/3) now.

**Comment by nicola** [comment]
See it in use here: https://github.com/linkeddata/ldnode/pull/164/files#diff-073738e0497d0d53d062fb1ab5567dbfR278

and it should return at least: `cert.modulus` and `cert.exponent` (public)

**Comment by sible777** [comment]
@nicola Could the method return a cert param to it's callback that contains an object literal representation of the certificate and a particular format of the certificate to be sent back to the client for installation?

Maybe something like below.

```javascript
var cert = {
    modulus: '...',
    exponent: '...',
    /*other fields here*/
    pkcs12: // or whatever format is needed here
}
```

**Comment by nicola** [comment]
Yes, you are right. Let's try and match gold: https://github.com/linkeddata/gold/blob/7bc6743db810b2a0d385ed5a84ea46ad1c1ed0c6/spkac.go#L109

**Comment by sible777** [comment]
Cool, thanks @nicola. I will work on this more later today :smile:
